sos/report/plugins/dellrac.py

```diff
@@ -0,0 +1,63 @@
+# This file is part of the sos project: https://github.com/sosreport/sos
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# version 2 of the GNU General Public License.
+#
+# See the LICENSE file in the source distribution for further information.
+
+from sos.report.plugins import Plugin, IndependentPlugin, PluginOpt
+import re
+import time
+
+
+class DellRAC(Plugin, IndependentPlugin):
+
+    short_desc = 'Dell Remote Access Controller Administration'
+
+    plugin_name = 'dellrac'
+    profiles = ('system', 'storage', 'hardware',)
+    packages = ('srvadmin-idracadm7',)
+
+    option_list = [
+        PluginOpt('debug', default=False, desc='capture support assist data')
+    ]
+
+    racadm = '/opt/dell/srvadmin/bin/idracadm7'
+
+    def setup(self):
+        for subcmd in ['getniccfg', 'getsysinfo']:
+            self.add_cmd_output(
+                '%s %s' % (self.racadm, subcmd),
+                suggest_filename='racadm_%s' % subcmd)
```

Comment:
Since the binary is ultimately `idracadm7`, we should use that in the filename. Otherwise users could be confused and think they are in fact running `racadm` here.

Reply:
The Dell documents refer to `racadm` command. There are other versions idracadm8 and 9 for other platforms where the RPM creates a symlink racadm. If and when the plugin supports those platforms or newer RPMs, then we want to make sure we be consistent with the filename. So, I would like to keep the output filename agnostic of the version.

Reply:
If there are symlinks dropped, then use call the symlink. If not, then we shouldn't be conflating the command that is run. If I see `racadm_getniccfg` then I assume the command that generated the output was exactly `racadm getniccfg`, not `idracadm7 getniccfg` or `idracadm8 getniccfg`. If I need to run one of the latter commands, then looking at `sos_commands/dellrac/racadm_*` does not track.

Reply:
The Dell guide refers to `racadm` everywhere. There are different versions of the RPM with the same usage and purpose. The (manual/automated) consumer of the data should not bother about whether the command was symlink, racadm, or versioned. There needs to be consistency in the output filename so that regardless of how the data was generated, the consumers are not impacted now and in future. If it helps, the plugin can be renamed to `racadm`. But I think having vendor name in the plugin name is apt.

Reply:
If the Dell guide refers to `racadm` - then this should be calling `racadm` if you want `racadm_$cmd` in the plugin directory. If there are different versions of the command out there then that is valuable information to pass on to whomever is reviewing the collected data. 

`suggest_filename` is not meant to obfuscate what is being run. It's used to make unruly or human-unfriendly command strings more readable.

Either actually call `racadm` here, or don't override the binary string in the resulting filename.

Reply:
The command will change to racadm with v8 and v9 support. The usage and output will remain same. Can we live with this until then?

Reply:
No.

As I've been saying we do not use `suggest_filename` to hide what we're executing from users. If we're running `idracadm7`, then the file here should be called `idracadm7_$cmd`. If we're running `racadm`, then at that point the file should be called `racadm_$cmd`.

What would be acceptable is to have a static _tag_ for this collection. You can add tags via the `tags` parameter (accepts `str` or `list` values), and those tags will appear in the manifest for this/these collections.

Reply:
Updated